### ISSUE TITLE
Signature Verification Failing

### ISSUE THREAD

**Issue by puneetmakkar** [yaronn/xml-crypto#137]
@yaronn @bjrmatos @danieljoppi 

Signature verfication fails here:
https://github.com/yaronn/xml-crypto/blob/master/lib/signed-xml.js#L280

**CanonicalizationMethod Algorithm**
http://www.w3.org/TR/2001/REC-xml-c14n-20010315#WithComments

**Public Key:**
```
-----BEGIN PUBLIC KEY-----
MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAgFhTDtuYJ5G5LEHMesnf
wX9cXZ1b/ozkkpbgtC3ziETiFkOFncbPCskpdbPmXxXv3vrwJ2RQIL2LZLZPe1xT
AyQY1DdD8hGqIemMwV2NqfFoEomVL5+QOAKCRiHkGgte6a2+OoTk9JzRP/NVaPkB
sdX1/nIPERYen3uDvUSYq83Ite2oDyaZZxj+/r46SadS/g5jWmeqgVoInJw813y7
Ee2HgYVbnktlLNhqIGj+1OKmwop+GP7Kk5CAkt9fo4VjRRllDaX1yFCZEbDL254n
S+LVOhl4mLBM8764+YVxjyYRC1Nq2rNZfQ602652i+l8u8nGqdiKOKDpjNDvhONP
yQIDAQAB
-----END PUBLIC KEY-----
```

**Signature:**
`aGtyMfcFIc9/Exl/S0tt5l8LeMM8ow3xvQrnSbAhJESWNSQsGYoRYf4y4Ypb7fpDA68boq15CzJPt6aC+LtGDyx5i/QF8M2QIwu5Ts3/mg0hGCjG+1JTfX9L8MAxYr4g38fE49JE9RoUrTybtRnb2AnXpk38J//WYTjdJDo7JTeLnU/hQKglOemjJTmfvJq5IetTtNBegLi2SPqBUg+LuWma3n6iHHAEL25v/CaO/9YK93S6XJ74u0Z86TLSMDkn8zNoG3+vggsJMK9Ofi3ErBOW5fB8P85jzwj3opm3wI0RZkPpwunqZ8zpIcGLztdzT8qfRg9Z+No4Ztj5jG4s3Q==`

**sigEncoding:**
`base64`

**signedInfoCanon**
`<SignedInfo xmlns="http://www.w3.org/2000/09/xmldsig#"><CanonicalizationMethod Algorithm="http://www.w3.org/TR/2001/REC-xml-c14n-20010315#WithComments"></CanonicalizationMethod><SignatureMethod Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"></SignatureMethod><Reference URI="#_b038441620b8e47b6c31dbc438f76c46"><Transforms><Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"></Transform></Transforms><DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256"></DigestMethod><DigestValue>q2bl+OfOtNL92DSNGRbro0CdVbaHc9WZKFJ6pSDUBls=</DigestValue></Reference></SignedInfo>`

Failing at the node's crypto verify call
<img width="1440" alt="signature_verification_fails" src="https://user-images.githubusercontent.com/23442385/28139503-5a5edb12-6709-11e7-80ba-ef8741f92231.png">

I am using https://github.com/tngan/samlify which internally uses xml-crypto to verify signatures.
I am able to verify SAML response and signature here https://www.samltool.com/validate_response.php

Any idea what could be happening ? Let me know if you need any more info.

**Comment by danieljoppi** [comment]
show the error

**Comment by puneetmakkar** [comment]
@danieljoppi  - It just returns false here https://github.com/yaronn/xml-crypto/blob/master/lib/signed-xml.js#L280

**Comment by puneetmakkar** [comment]
@danieljoppi 

**SAML Response**
<?xml version="1.0"?>
<saml2p:Response xmlns:saml2p="urn:oasis:names:tc:SAML:2.0:protocol" Destination="https://dev-memberportal.lifelock.com/nsl/response" ID="_b038441620b8e47b6c31dbc438f76c46" InResponseTo="_3852c847-a8a9-4c33-9dca-9a984a74f52f" IssueInstant="2017-07-12T19:37:09.091Z" Version="2.0">
  <saml2:Issuer xmlns:saml2="urn:oasis:names:tc:SAML:2.0:assertion">https://login-int.norton.com/sso/idp/SAML2</saml2:Issuer>
  <Signature xmlns="http://www.w3.org/2000/09/xmldsig#">
    <SignedInfo>
      <CanonicalizationMethod Algorithm="http://www.w3.org/TR/2001/REC-xml-c14n-20010315#WithComments"/>
      <SignatureMethod Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"/>
      <Reference URI="#_b038441620b8e47b6c31dbc438f76c46">
        <Transforms>
          <Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"/>
        </Transforms>
        <DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256"/>
        <DigestValue>q2bl+OfOtNL92DSNGRbro0CdVbaHc9WZKFJ6pSDUBls=</DigestValue>
      </Reference>
    </SignedInfo>
    <SignatureValue>aGtyMfcFIc9/Exl/S0tt5l8LeMM8ow3xvQrnSbAhJESWNSQsGYoRYf4y4Ypb7fpDA68boq15CzJPt6aC+LtGDyx5i/QF8M2QIwu5Ts3/mg0hGCjG+1JTfX9L8MAxYr4g38fE49JE9RoUrTybtRnb2AnXpk38J//WYTjdJDo7JTeLnU/hQKglOemjJTmfvJq5IetTtNBegLi2SPqBUg+LuWma3n6iHHAEL25v/CaO/9YK93S6XJ74u0Z86TLSMDkn8zNoG3+vggsJMK9Ofi3ErBOW5fB8P85jzwj3opm3wI0RZkPpwunqZ8zpIcGLztdzT8qfRg9Z+No4Ztj5jG4s3Q==</SignatureValue>
    <KeyInfo>
      <X509Data>
        <X509Certificate>MIIDjTCCAnWgAwIBAgIEE+Z7jDANBgkqhkiG9w0BAQsFADB3MQswCQYDVQQGEwJVUzELMAkGA1UECBMCQ0ExFDASBgNVBAcTC0N1bHZlciBDaXR5MR0wGwYDVQQKExRTeW1hbnRlYyBDb3Jwb3JhdGlvbjEMMAoGA1UECxMDTlNMMRgwFgYDVQQDEw9IaW1hbnNodSBTaGFybWEwHhcNMTcwMTIwMTcwNjQwWhcNMjAwMTIwMTcwNjQwWjB3MQswCQYDVQQGEwJVUzELMAkGA1UECBMCQ0ExFDASBgNVBAcTC0N1bHZlciBDaXR5MR0wGwYDVQQKExRTeW1hbnRlYyBDb3Jwb3JhdGlvbjEMMAoGA1UECxMDTlNMMRgwFgYDVQQDEw9IaW1hbnNodSBTaGFybWEwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCAWFMO25gnkbksQcx6yd/Bf1xdnVv+jOSSluC0LfOIROIWQ4Wdxs8KySl1s+ZfFe/e+vAnZFAgvYtktk97XFMDJBjUN0PyEaoh6YzBXY2p8WgSiZUvn5A4AoJGIeQaC17prb46hOT0nNE/81Vo+QGx1fX+cg8RFh6fe4O9RJirzci17agPJplnGP7+vjpJp1L+DmNaZ6qBWgicnDzXfLsR7YeBhVueS2Us2GogaP7U4qbCin4Y/sqTkICS31+jhWNFGWUNpfXIUJkRsMvbnidL4tU6GXiYsEzzvrj5hXGPJhELU2ras1l9DrTbrnaL6Xy7ycap2Io4oOmM0O+E40/JAgMBAAGjITAfMB0GA1UdDgQWBBRkFxGn2eznUl5Zd0ojSThY5fd/PzANBgkqhkiG9w0BAQsFAAOCAQEAKKw7hB0H5zqW1vwJ/OzJNV1o44aoS6/vY2570E8DIgrsqoAuiulCbfQetqPcbleVx8TvQ1GCvb/o37LEQA6NgYuZsB43z+vV+ZYVy7b5jbdxsfJ7tsybUdnEoK+f+p4pQRTgVqS/46P0cv+kuDTJ5urk1jfhl9PL5wDT6HVIafVPcXLAqrb+M/Xb3Asaj4Fasx5r7E8IuqiSDPOr5PTrlyWlBZaO2aVU2V49zbIygtE9x5VobxIBWNv8e1G9Dekge3GD7jn9jsFfnM0kdsq3bGgRIsRNaauNy9vPIkPhYEAXzZrU+WQRJVlM5dvJ6MtfgU9j3eCyC+Z/JsR1lwOBhA==</X509Certificate>
      </X509Data>
    </KeyInfo>
  </Signature>
  <saml2p:Status>
    <saml2p:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success"/>
  </saml2p:Status>
  <saml2:EncryptedAssertion xmlns:saml2="urn:oasis:names:tc:SAML:2.0:assertion">
    <xenc:EncryptedData xmlns:xenc="http://www.w3.org/2001/04/xmlenc#" Id="_590ffbf61de253125f6bf578a7b256a1" Type="http://www.w3.org/2001/04/xmlenc#Element">
      <xenc:EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#aes256-cbc"/>
      <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
        <ds:RetrievalMethod Type="http://www.w3.org/2001/04/xmlenc#EncryptedKey" URI="#_024121007cd43b6f0fb5ae067d6b30e8"/>
      </ds:KeyInfo>
      <xenc:CipherData>
        <xenc:CipherValue>3qmh29izpJKeGrNX0gCQm1IW8Bpzo/aPi72wib9t+glFXjb9UexOIKQTzxh/XHtdy1PMB/DfJBRG6UjM4TnDvuX3yfSPgA5mSrd9Ao6mcRfrOzakQivbZ1XNf/uQsWp8p0uOMi8lDizd/m5Q1/T2TCyLeQw1dwrPF3LsDN3tE7HADDOwFDnnDS7JtT8Pm0o8YwpAL0kxRKFQz+412Chh9JbPOkN3aR1ukjF7/LE1HoOB5zohLei6JHB124Gz7hGFcC2R+0fYfF6DcItkP7rQEHNjxD3CaR2swRB8HpnnXTLOwCLmTx9jwdCgzxrOcK1CqZ/uOI/+pBugl5HAhcPYmGsS4GeotQhjQSKqO+Sq6PUV/W2Khw9gCWQkjRcFQmTSKzeNMtkRlY4hLGcjhs5oZlOPPvWttWqQnNhxXaMixk9P4T3En8bAjKXQAW3dmLIgUTnJ5fsAuUZfEcD/gCIDTjw5imKGCaTrZnVMQ//86TJGwEFQoBLpowoTlFjwHgwl/8ap2B1wD4zWllxK2blYaqs91h6gpKOVAcYG8wx89BPFcwUbKrWmt7la1Zte15iRSlEOCpGxAIvgNXpwkhI+sR7DNTwAipt/LrvnhePEt9SbrH94AXGNC43BMhMiw29FMgE2kKYRe4+qvOJq9AsDsrp7ReHHtYESTx9CDkGd9UQEdFGf+RUQ2xti+JUfX4Wz/a4OYqsDhc9sK77G4Wj5tEbgWDBtb7N2MSH+VsaAA9T7x8+bLLo7IGSsIqOPoPk9IutBj8lXb30AyV/MEVT+6puG6bllieDOnb2tsiYoqrc2d1epVpKM1JKakBSaM8GonKiA/wIg/44wlUrVOmznxbZNkqSkbtxmT5xg3Rn1mtV0ZpKRCfVd+dsPh3hogw8bAYpTAgPIgCJLL4y6GVjAoSMOrQ60UaEU+Dbsk4s0FkzD8qQovRA0Y8cungUklZz7lzM1LtkYr37vnu52f2NeVSbrP0as6iHZFR7r2UFmExI0wSHJGlWFEtPt+Dwsp2jyHat6dg7T2gSBZ1DUu//am3Yn3znsoQT5fDfy99YQ+516Yuf+5DUfcAmjZUhRN9t+EYrggNLo7djkMrSJeDe2lAyGbgIOrwyCI93WRHRKf10uQtH+v3hkdwAXEy5Tvw3aS8pdD3SfQW7arGe7MXw2mHmApL7mpSYjfa8rI4LkJmbg4A+RiLdsoOG//x2O2Ip8b7Jix/G1jTZ/K8NPr8zgi6xRV2ZWpYyeJKMGPFd+aHFCMzbaU9eMG4kYMK2APtUezmGDKHthsYlV80sUEUomXpF+C3tr3OHH3L6cv9SVrho+8nXoKlD5kYZVyq1YshhL2LES+m5qLe/EoQtgEfgjDSwap67/cTT3bSKCMrD58jWsfyFPoT7Jnedk4ADLUg/tXTfxDIge1aO+9x0Qnn3d7tsOKDIvPbnCVmjKS0gz279AtwhnBUin9yD6kJvZiRfI+p9jiiUU3IaC2NlCW4S8QKk6FOMSbKh/RvATad+/wS+sj7kbOPVRhiSmhUZHtXJck1DAyVDh0cij7GMx4nuofAJGKWS3O9F1x5x/j3FXpMHUU15uM90G1TZrXrgFDAxM+KbdKiE8NZCyArZvX6/cDrel0DBIPN8b07Ri0Hd2gFUNZ0BZviAi24OCA2ZHEp6c2cwtlYnzDgwJGT6QrkGmtcDHzkAdKHrbeZvacqlNiUnF+nEy+W74Q5WtHeBGyCKX+rxNwwxcoE2g77nupLXVoCVByMWsVwaZipiCFRQNVusEvxK3IFbCFMdBt/aAxsQc85CHUP5IwYZCyt86mrKSqD2v70bko94lxH0RcvITz4MLZyhtulqPPIj6wWZ9ce6jxDbJUBqltknrPe0M6Yp9B58bhuJU+NyBDbTtk48uQatt6RxGbJ+RgqoNegItNB5Cl7oTG5FUkWoM2fBp3tf8KOKJhToXXna2I7TYAs2YTmZ5fpfdljffciLuxGctgCY9DjM5FHKImco7MId04s5OmpNM0CLEl9zyoKliV//MY3DJPQidVnXJzuaL7GVSHzsKASC1VUKMSX3ZVfzkzS4y+1rrPBJTjiOFLvx0FzGqCLwxsks/uV8ZnDwMLiK28Nmcl7sBfMwPzRMgesayAlNOHrdQ9RshK30lPY7wInyLuvyGMuSuoZL+mSw2BRtSt6q5VTZ4cPvMOTSPX2uCyCQ5RGpHuq2ZLPdmZhg5WemUg4gqMyHK/9Kq2F2FKK5IBB1NDj7Hk2anhgpaSRRq1SMTaVy5xkcrYcDK6hDjR5DPL9yzgs8Z+0YGwofr66iLkl3YBBEGf0RQnDKVvQtdBeEaCtYgcg1RQP9h70ETDJWAVzK7u0ZDmmFFitg3fqoPv0fd0HENR6mB3PVTl7FTLI9XDWdR/uEtcAO1ZLc3fc3CHEzYywizqYS50CjcGiiFPCkeoaywXWryDKkjnIWFkH6bK0RvekPZxY0Ve2IjOi5Nvft8QMjoEU3J/pKCnsZqPuRN/chq2wEdwSwfYJitKeZyxnzvFjcPPD9Vop9+SGPE0Z0pXmyz3+I36wps3uaDy1/JD08Dgofp0lmLlJ5xVr1s7o/ijFdq1FRbMOdC5VwMrKaEuJV7MNE5WNYKictpqzsyXVeGH21d5ZLOXd86fZ31aelJ25IPrsmdC3LsCdDF1Cjr4TTslWaGSOPadNRnFqZX+973f7pt+MfICNqEo3RbkowURuEqtjbQHcXcLb4LBA1O+e9ack1qOAJwUBtZxv87yorIw5EIiJMfITaUwGA+pNFTYf5IV1zECVoXLLOH59UDYd5tc+bQJyy+0Sa1TTDMuU1VtS+BvwPIUXlDj84/5GlNEcNy33EA9vlSFiMz5/hZjHTYmWdoPnz4IQm3gvvph1oFJFXXdunqDwlp0ZoEF2cvE18ceq7CouzHq1+QjKBuQoEXoLJZXt0dY3q2Ke9rvmsXf5/BOpeEnqssSdn6xCxCyFSXTUOOQG6F2pwpoo9+0lsrlb6tk9RvIizTmIzEObfZa+DpZByKUV/v+Fy66APMmv2MlCFG7PA3z7kG2KxWmHaLLWOlg2cBJrRAuvCExkx1c4fi87ikcvEaj6TY8vvuZSjDFoSpqdo2W18lhGK0N4rbiS2//FHEpvQe14OJy/RhjhKpmxGxnbR7rheWoU1xrBqnE1OzD554B2j7RQGTBRJNQO4yC1Go4bgBU7FqkAleg3J3kKt8H4fDhWp4cMyZNUivXDt3NDNFPIKH/xVhlmNk2tMgccQ2M1VzW4WKH2nfXrNKWiZSPd1ybb6ehC9Xz+2karlwGdzspi3tWUr3Js2bj0+9ATc/NcNz7xqwuN3XmRFBCTtvMSihspqXz2V3kTjB639dyplCcI2yeFuZxQvLFbDu+byF2PkRDovV5DyMhcvqIHAdrr6sH/F1ZMTWysZzUxgPWJ8wguWn388CjIgqhBWSreD7VH+Z/tqC</xenc:CipherValue>
      </xenc:CipherData>
    </xenc:EncryptedData>
    <xenc:EncryptedKey xmlns:xenc="http://www.w3.org/2001/04/xmlenc#" Id="_024121007cd43b6f0fb5ae067d6b30e8">
      <xenc:EncryptionMethod Algorithm="http://www.w3.org/2001/04/xmlenc#rsa-1_5"/>
      <ds:KeyInfo xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
        <ds:X509Data>
          <ds:X509Certificate>MIIDbDCCAlQCCQCrUakZgqQW+zANBgkqhkiG9w0BAQUFADB4MQswCQYDVQQGEwJVUzETMBEGA1UE
CBMKQ2FsaWZvcm5pYTEWMBQGA1UEBxMNTW91bnRhaW4gVmlldzEWMBQGA1UEChMNU3ltYW50ZWMg
SW5jLjERMA8GA1UECxMITGlmZWxvY2sxETAPBgNVBAMTCGxpZmVsb2NrMB4XDTE3MDYyMzIyMTEz
MVoXDTE3MDcyMzIyMTEzMVoweDELMAkGA1UEBhMCVVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFjAU
BgNVBAcTDU1vdW50YWluIFZpZXcxFjAUBgNVBAoTDVN5bWFudGVjIEluYy4xETAPBgNVBAsTCExp
ZmVsb2NrMREwDwYDVQQDEwhsaWZlbG9jazCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEB
ANj/A11mLwS1SOGsvRs57QhzFBO11sSyR9KV2XkNXYqZQZgQyEqnuPMPQZmjmAVZ9RsD/2dYSci+
YohI5z1NNcEiJCksS9c7vYYJou97Wqbve0QmQoGkWNdlaGn0CFqArAL7FyxDUc3P4Nhw18Ueo42q
3+1YK0evpEr9iupMazq2B66EvA4+1Miqgj5P+5vptQpWA4soUTZtP85a13e6b+ytrU1IpyVldhB5
bwXIkrWyWOcZQ1S/lQx4TUMUkOGrY2GrEwmAtUINOZAFmKQ6mKGdcvYfmfjCXFFS0KV3Hci7wnPR
pN8RfA5XeI1wIoPjIpiGKsVwSAocuAmIguEKRrMCAwEAATANBgkqhkiG9w0BAQUFAAOCAQEAqQc5
prs+HlQLdXpCp6gA9QY3b64GuBFWsnQAQ0ORDSK5qAw9oY/KrsmAzI+kv2QMJxExcAp7MFRQnNne
Ii+wGjk4hQ2VwMakdfx4G7ymGGORJgxGXPDngQXm+I2h0sbKkWZXnlGeoxf/P8PIkOrf+0nY8YBR
ElNHO3IcnddRi6IEwe9MBpO8wGgEfq/ILOM5kkUhCrjiDWjhdeIvrEOIa9ZIu4cZEA5TzI+W2C57
FPJX0OdoseoIh0XhK3B5BpB2K68HUEkWPYEFvc84ymwce32OJdF+5tFtfFrICDDCxFKXVPXkkIVh
kBif3q1s8E/fnkWFdwZxZlvb5GZzJKoqGw==</ds:X509Certificate>
        </ds:X509Data>
      </ds:KeyInfo>
      <xenc:CipherData>
        <xenc:CipherValue>hog9p/no/RBeegMLVQGJrk7T27ugxwEU+ahhkNdITExLRhrzJxDX54nZppJ18Kj0PilgcFbvvJIFKex+CxsSMBf94ySUuAAfVcxsiy5lgLJccF3lrPVFx789CYEXA65Q4eniruhMLjDDZmAf0zINtFPkztYMKjza4Hz0HXe2XOSx2f/OpuaieySM/7+xSl/tRC6VizuYfndGB7dQh+7ZrVfxOdG/x21UAMXRQpm/fj8Mm8WmgX+k8exVEHwkzNKzUjvfYzrcdFvZo2uKnFLnbUsiQaq+0LgH9Zibzqq+E90OnsoXeU7Itkb28ltMbMv8ysZRVrs7V2FB39PQR2qfsQ==</xenc:CipherValue>
      </xenc:CipherData>
      <xenc:ReferenceList>
        <xenc:DataReference URI="#_590ffbf61de253125f6bf578a7b256a1"/>
      </xenc:ReferenceList>
    </xenc:EncryptedKey>
  </saml2:EncryptedAssertion>
</saml2p:Response>


---


**Partner Signs with the following xml**
<SignedInfo xmlns="http://www.w3.org/2000/09/xmldsig#" xmlns:saml2p="urn:oasis:names:tc:SAML:2.0:protocol">
  <CanonicalizationMethod Algorithm="http://www.w3.org/TR/2001/REC-xml-c14n-20010315#WithComments"/>
  <SignatureMethod Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"/>
  <Reference URI="#_b038441620b8e47b6c31dbc438f76c46">
    <Transforms>
      <Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"/>
    </Transforms>
    <DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256"/>
    <DigestValue>q2bl+OfOtNL92DSNGRbro0CdVbaHc9WZKFJ6pSDUBls=</DigestValue>
  </Reference>
</SignedInfo>


------
**xml's crypto cannonxml**

<SignedInfo xmlns="http://www.w3.org/2000/09/xmldsig#">
  <CanonicalizationMethod Algorithm="http://www.w3.org/TR/2001/REC-xml-c14n-20010315#WithComments"/>
  <SignatureMethod Algorithm="http://www.w3.org/2001/04/xmldsig-more#rsa-sha256"/>
  <Reference URI="#_b038441620b8e47b6c31dbc438f76c46">
    <Transforms>
      <Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature"/>
    </Transforms>
    <DigestMethod Algorithm="http://www.w3.org/2001/04/xmlenc#sha256"/>
    <DigestValue>q2bl+OfOtNL92DSNGRbro0CdVbaHc9WZKFJ6pSDUBls=</DigestValue>
  </Reference>
</SignedInfo>

------

Partner's signing xml has additional "xmlns:saml2p="urn:oasis:names:tc:SAML:2.0:protocol" in the Signed info compared to xml's cryptos cannon xml.

How can we fix this ?

**Comment by Hinaser** [comment]
I sent PR which fixed non-exclisive c14n issue.
I believe my PR resolve your issue.
#157

**Comment by LoneRifle** [comment]
This has been merged as part of #161 and released as part of the 1.x series. As Hinaser says, this should be fixed by now. If it isn't, please re-open.
